src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/osd/osd-recv-speed-modal/osd-recv-speed-modal.component.ts

```diff
@@ -76,24 +80,14 @@ export class OsdRecvSpeedModalComponent implements OnInit {
   }
 
   ngOnInit() {
-    const observables = [];
-    Object.keys(this.priorityAttrs).forEach((configOptionName) => {
-      observables.push(this.configService.get(configOptionName));
-    });
-
-    observableForkJoin(observables)
-      .pipe(
-        mergeMap((configOptions) => {
-          return of(this.getCurrentValues(configOptions));
-        })
-      )
-      .subscribe((resp) => {
-        this.detectPriority(resp.values, (priority) => {
-          this.setPriority(priority);
-        });
-        this.setDescription(resp.configOptions);
-        this.setValidators(resp.configOptions);
+    this.configService.filter(Object.keys(this.priorityAttrs)).subscribe((data: any) => {
```

Comment:
https://github.com/ceph/ceph/pull/27072/files#diff-c5e0124e8b8d21592f4416642d1504f3R66 ?

Reply:
I mend the code in this file, the function that is called when subscribed to more exact. Sorry for the confusion.

Reply:
https://github.com/s0nea/ceph/blob/wip-dashboard-pg-scrub/src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/osd/osd-recv-speed-modal/osd-recv-speed-modal.component.spec.ts#L195

https://github.com/s0nea/ceph/blob/wip-dashboard-pg-scrub/src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/osd/osd-recv-speed-modal/osd-recv-speed-modal.component.spec.ts#L124

https://github.com/s0nea/ceph/blob/wip-dashboard-pg-scrub/src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/osd/osd-recv-speed-modal/osd-recv-speed-modal.component.spec.ts#L70

https://github.com/s0nea/ceph/blob/wip-dashboard-pg-scrub/src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/osd/osd-recv-speed-modal/osd-recv-speed-modal.component.spec.ts#L237

https://github.com/s0nea/ceph/blob/wip-dashboard-pg-scrub/src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/osd/osd-recv-speed-modal/osd-recv-speed-modal.component.spec.ts#L252

??

Reply:
@p-na As discussed during a chat, I don't like to call it that it's more polite calling it humble. Also the reason that I request the changes makes it clear that I would like to see the change happening.

@s0nea I've looked into the tests, thought it would be very easy to write, but it's a bit more complicated than thought. I will create a separate PR to test the init function, if that is OK for you.